x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/logic/result_settings_logic.test.ts

```diff
@@ -0,0 +1,412 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { LogicMounter } from '../../../../__mocks__';
+
+import { Schema, SchemaConflicts, SchemaTypes } from '../../../../shared/types';
+import { OpenModal, ServerFieldResultSettingObject } from '../types';
+
+import { ResultSettingsLogic } from '.';
+
+describe('ResultSettingsLogic', () => {
+  const { mount } = new LogicMounter(ResultSettingsLogic);
+
+  const DEFAULT_VALUES = {
+    dataLoading: true,
+    saving: false,
+    openModal: OpenModal.None,
+    nonTextResultFields: {},
+    resultFields: {},
+    serverResultFields: {},
+    textResultFields: {},
+    lastSavedResultFields: {},
+    schema: {},
+    schemaConflicts: {},
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const itShouldCloseOpenModals = (callback: () => void) => {
+    mount({
+      openModal: OpenModal.ConfirmSaveModal,
+    });
+
+    callback();
+
+    expect(ResultSettingsLogic.values).toEqual({
+      ...DEFAULT_VALUES,
+      openModal: OpenModal.None,
+    });
+  };
```

Comment:
An alternative / more "test"-y way of writing this, without a fn/callback:

```js
describe('actions',
  // ...

  describe('actions that close open modals', () => {
    beforeEach(() => {
      mount({
        openModal: OpenModal.ConfirmSaveModal,
      });
    });

    afterEach(() => {
      expect(ResultSettingsLogic.values).toEqual({
        ...DEFAULT_VALUES,
        openModal: OpenModal.None,
      });
    });

    it('closeModals', () => ResultSettingsLogic.actions.closeModals());
    it('resetAllFields', () => ResultSettingsLogic.actions.resetAllFields());
  });
```

up to you whether you like it more or less 🤷 Since there's only 2 instances of this also I wouldn't mind just repeating the code honestly

Reply:
f3e126ef2bb2764e209b147eb2b483d163f5c555